src_c/scrap.c

```diff
@@ -355,6 +408,76 @@ _scrap_set_mode(PyObject *self, PyObject *args)
 
 #endif /* !defined(MAC_SCRAP) */
 
+/**
+ * @brief Fetches a python string from the SDL clipboard. If
+ *        there is nothing in the clipboard, it will return empty
+ *
+ * @return PyObject*
+ */
+static PyObject *
+_scrap_get_text(PyObject *self, PyObject *args)
+{
+    char *text = SDL_GetClipboardText();
+
+    if (!text) {
+        const char *err = SDL_GetError();
+        if (err) {
+            SDL_free(text);
+            PyErr_SetString(PyExc_RuntimeError, err);
+            return NULL;
+        }
+    }
+
+    PyObject *returnValue = Py_BuildValue("s", text);
```

Comment:
Nitpick- but this seems like a strange way of doing this.

This would work iirc: https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_FromString